guardian/support-dotcom-components
# Add ability to retrieve second banner channel from the banner sharing tool

## What does this change?
This adds a new BannerTestGenerator function that retrieves data from the 'banner 2' channel set up in [support-admin-console](https://github.com/guardian/support-admin-console/pull/156) and formats it for use with the Digital Subscriptions and Guardian Weekly banners. This is currently behind an env variable switch, so won't affect production until we're ready to turn it on.

## How to test
Run the server with `stage=CODE yarn dev`, or on CODE, to enable retrieving the second banner channel data. POST-ing the following to /banner should return a result indicating the digital subscriptions banner, but with data drawn from the second channel file in Code

```
{
    "tracking": {
        "ophanPageId": "kf2myg62kgw94ha9e3fv",
        "platformId": "GUARDIAN_WEB",
        "clientName": "frontend",
        "referrerUrl": "https://www.theguardian.com/uk"
    },
    "targeting": {
        "alreadyVisitedCount": 3,
        "isPaidContent": false,
        "showSupportMessaging": true,
        "engagementBannerLastClosedAt": "2020-09-15T16:36:54.873Z",
        "mvtId": 987471,
        "countryCode": "GB",
        "switches": {
            "remoteSubscriptionsBanner": true
        },
        "weeklyArticleHistory": [
            {
                "week": 18511,
                "count": 5
            }
        ]
    }
}
```
